src/services/sam.ts

```diff
@@ -99,11 +99,12 @@ export const selectEffectiveCost = (validator: AuctionValidator) => (validator.m
 
 export const bondColorState = (validator: AuctionValidator, samDistributedStake: number, maxMarinadeTvlSharePerValidatorDec: number): Color => {
     const maxValidatorStakeShare = maxMarinadeTvlSharePerValidatorDec * samDistributedStake
-    const stake = validator.maxStakeWanted >= maxValidatorStakeShare ? maxValidatorStakeShare : validator.maxStakeWanted
-    const bondReq = bondBalanceRequiredForStakeAmount(stake, validator)
-    if (validator.bondBalanceSol > bondReq * 2) {
+    const stake = validator.auctionStake.marinadeSamTargetSol >= maxValidatorStakeShare ? maxValidatorStakeShare : validator.auctionStake.marinadeSamTargetSol
```

Comment:
(non-blocking)
I'm not 100% sure whether there couldn't be some edge cases (so feel free to ignore my comment and play it safe), but if there is no max stake wanted now then `marinadeSamTargetSol` should never be higher than `maxValidatorStakeShare` for any validator (otherwise the auction would provide results in conflict with the defined rules), so using just `validator.auctionStake.marinadeSamTargetSol` directly should be enough here.